src/UI/Art/SpriteArtist/SpriteLibrary.java

```diff
@@ -288,17 +278,103 @@ public static Sprite getMoveCursorArrowSprite()
+      buildingColorPalettes.put(Color.PINK, new ColorPalette(pinkMapBuildingColors));
+      buildingColorPalettes.put(Color.CYAN, new ColorPalette(cyanMapBuildingColors));
+      buildingColorPalettes.put(Color.ORANGE, new ColorPalette(orangeMapBuildingColors));
+      buildingColorPalettes.put(PURPLE, new ColorPalette(purpleMapBuildingColors));
+
+      mapUnitColorPalettes.put(Color.PINK, new ColorPalette(pinkMapUnitColors));
+      mapUnitColorPalettes.put(Color.CYAN, new ColorPalette(cyanMapUnitColors));
+      mapUnitColorPalettes.put(Color.ORANGE, new ColorPalette(orangeMapUnitColors));
+      mapUnitColorPalettes.put(PURPLE, new ColorPalette(purpleMapUnitColors));
+
+      // Throw some color names in there for the defaults
+      // toString() is not user-friendly
+      colorNames.put(Color.PINK, "salmon");
+      colorNames.put(Color.CYAN, "cyan");
+      colorNames.put(Color.ORANGE, "citrus");
+      colorNames.put(PURPLE, "sparking");
 
-  private static ColorPalette getBuildingColors(Color colorKey)
+      // We want to be able to use the normal units, as well as any others
+      factionNames.add(DEFAULT_SPRITE_KEY);
+
+      final File folder = new File("res/unit/faction");
+
+      if (folder.canRead())
+      {
+        for( final File fileEntry : folder.listFiles() )
+        {
+          // If it's a file, we assume it's a palette
+          if( !fileEntry.isDirectory() )
+          {
+            String colorName = fileEntry.getAbsolutePath();
+            if( colorName.contains(".png") )
+            {
+              BufferedImage bi = loadSpriteSheetFile(colorName);
+              // Grab the last color on the first row as our "banner" color
+              Color key = new Color(bi.getRGB(bi.getWidth() - 1, 0));
+              Color[] cUnits = new Color[defaultMapColors.length];
+              Color[] cStructs = new Color[defaultMapColors.length];
+              for( int i = 0; i < defaultMapColors.length; i++ )
+              {
+                cUnits[i] = new Color(bi.getRGB(i, 0));
+                cStructs[i] = new Color(bi.getRGB(i, 1));
+              }
+              buildingColorPalettes.put(key, new ColorPalette(cStructs));
+              mapUnitColorPalettes.put(key, new ColorPalette(cUnits));
+              colorNames.put(key, fileEntry.getName().replace(".png", ""));
+            }
+          }
+          else // If it's a directory, we assume it's a set of map sprites, i.e. a faction.
+          {
+            factionNames.add(fileEntry.getName());
+          }
+        }
+      }
+    }
+  }
+
+  public static ColorPalette getBuildingColors(Color colorKey)
   {
+    initResources();
     return buildingColorPalettes.get(colorKey);
   }
 
   public static ColorPalette getMapUnitColors(Color colorKey)
   {
+    initResources();
     return mapUnitColorPalettes.get(colorKey);
   }
 
+  public static String getColorName(Color colorKey)
+  {
+    initResources();
+    return colorNames.get(colorKey);
```

Comment:
Do you think it's worth raising some kind of flag if for some reason `colorKey` is not in `colorNames`?

Reply:
I added a default to salmon and... default... in the test branch.
We don't really have any sort of proper "hey, user, you did a wrong thing" other than stuff going weird or crashing, yet.
That's a system that probably should exist, but it's not super-useful to me at this time and I don't have any ideas.

Reply:
For something like this that should (in theory) be less error-prone/more visible when it fails, maybe this is fine, but I think I'd prefer at least a printed warning and a default like you mentioned. If/when we want something more robust, we can add it.